File: quote/Projects/PluginModel/ActiveHeader.vb
Imports System.ComponentModel

Imports Model.Common

Public Class ActiveHeader
    Implements INotifyPropertyChanged

    Private _Header As Header

    Private Shared ReadOnly _ActiveHeader As ActiveHeader = New ActiveHeader

    Public Shared ReadOnly Property ActiveHeader As ActiveHeader 
        Get
            Return _ActiveHeader
        End Get
    End Property

    Public Event PropertyChanged As PropertyChangedEventHandler _
        Implements INotifyPropertyChanged.PropertyChanged

    Public Property Header As Header
        Get
            Return _Header
        End Get
        Set(ByVal value As Header)
            If (_Header IsNot value) Then
                _Header = value
                RaiseEvent PropertyChanged(Me, _
                    New PropertyChangedEventArgs("Header"))
            End If
        End Set
    End Property

End Class

File: quote/Projects/PluginModel/ActiveHeader.cs
namespace Model
{
    using System;
    using System.ComponentModel;
    using System.Linq;
    
    using Model.Common;

    /// <summary>
    /// Holds the <c>header</c> that is currently active.
    /// The class is a singleton.
    /// </summary>
    public class ActiveHeader : INotifyPropertyChanged
    {
        /// <summary>
        /// The only instance.
        /// </summary>
        private static ActiveHeader instance = new ActiveHeader();
        
        /// <summary>
        /// The active header.
        /// </summary>
        private Header header;

        /// <summary>
        /// Event used to signal the header has changed.
        /// </summary>
        public event PropertyChangedEventHandler PropertyChanged;

        /// <summary>
        /// Gets the singleton.
        /// </summary>
        /// <value>The active header.</value>
        public static ActiveHeader Instance 
        {
            get { return instance; }
        }

        /// <summary>
        /// Gets or sets the active <c>header</c>.
        /// </summary>
        /// <value>The header.</value>
        public Header Header 
        {
            get 
            { 
                return this.header; 
            }
            
            set 
            {
                if (!object.ReferenceEquals(this.header, value))
                {
                    this.header = value;
                    
                    if (this.PropertyChanged != null) 
                    {
                        this.PropertyChanged(this, new PropertyChangedEventArgs("Header"));
                    }
                }
            }
        }
    }
}